sphinx/util/osutil.py

```diff
@@ -174,7 +174,7 @@ class FileAvoidWrite:
     Objects can be used as context managers.
     """
 
-    def __init__(self, path: str) -> None:
+    def __init__(self, path: Path) -> None:
```

Comment:
This should be `str | os.PathLike[str]` technically.

Reply:
> This should be `str | os.PathLike[str]` technically.

why can't it be changed to `Path`?

Reply:
Technically, this one does not require to be a `Path` object and just a str or a path-like object is fine (according to its usage). Since we typed it to be a `str`, it's better if we don't type it with an incompatible type to avoid mistakes (and possible downstream issue I'd say, though I'm not sure whether this object is part of the public API or not outside of Sphinx).

Reply:
> Since we typed it to be a str, it's better if we don't type it with an incompatible type to avoid mistakes

changing the stringly-typed file handling to use a consistent type is kind of the whole point of this PR.

basically to change all of the `str | os.Pathlike[str]` calls to `pathlib.Path`